tools/c7n_azure/c7n_azure/policy.py

```diff
@@ -107,6 +107,11 @@ def __init__(self, policy):
         location = self.service_plan.get('location', 'westus2')
         rg_name = self.service_plan['resource_group_name']
 
+        target_sub_id = \
+            local_session(self.policy.session_factory).get_function_target_subscription_id()
```

Comment:
Unfortunately you can't do this in constructor.. It will try to authenticate with Azure, right? 
(try custodian validate)